Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -1143,33 +1145,54 @@ void SoftwareGLContext::gl_read_pixels(GLint x, GLint y, GLsizei width, GLsizei
         return static_cast<GLuint>(0xffffffff * min(max(f, 0.0f), 1.0f));
     };
 
+    u8 component_size = 0;
+    switch (type) {
+    case GL_BYTE:
+    case GL_UNSIGNED_BYTE:
+        component_size = 1;
+        break;
+    case GL_SHORT:
+    case GL_UNSIGNED_SHORT:
+        component_size = 2;
+        break;
+    case GL_INT:
+    case GL_UNSIGNED_INT:
+    case GL_FLOAT:
+        component_size = 4;
+        break;
+    }
+
     if (format == GL_DEPTH_COMPONENT) {
+        // Align rows according to pack alignment
+        auto const component_alignment = max(component_size, m_pack_alignment / component_size);
+        auto const aligned_width = (width + component_alignment - 1) / component_alignment * component_alignment;
+
         // Read from depth buffer
         for (GLsizei i = 0; i < height; ++i) {
             for (GLsizei j = 0; j < width; ++j) {
                 float depth = m_rasterizer.get_depthbuffer_value(x + j, y + i);
 
                 switch (type) {
                 case GL_BYTE:
-                    reinterpret_cast<GLchar*>(pixels)[i * width + j] = float_to_i8(depth);
+                    reinterpret_cast<GLchar*>(pixels)[i * aligned_width + j] = float_to_i8(depth);
                     break;
                 case GL_SHORT:
-                    reinterpret_cast<GLshort*>(pixels)[i * width + j] = float_to_i16(depth);
+                    reinterpret_cast<GLshort*>(pixels)[i * aligned_width + j] = float_to_i16(depth);
```

Comment:
Referencing my comment above, I would prefer this instead:
```suggestion
                    *reinterpret_cast<GLshort*>(char_ptr) = float_to_i16(depth);
```